src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -1360,6 +1457,88 @@ private Object writeReplace() {
             return new CollSer(CollSer.IMM_MAP, array);
         }
     }
+
+    static final class StableMap<K, V>
+            extends AbstractImmutableMap<K, V> {
+
+        @Stable
+        private final Function<? super K, ? extends V> mapper;
+        @Stable
+        private final Map<K, StableValueImpl<V>> delegate;
+
+        StableMap(Set<K> keys, Function<? super K, ? extends V> mapper) {
+            this.mapper = mapper;
+            this.delegate = StableValueFactories.map(keys);
+        }
+
+        @Override public boolean              containsKey(Object o) { return delegate.containsKey(o); }
+        @Override public int                  size() { return delegate.size(); }
+        @Override public Set<Map.Entry<K, V>> entrySet() { return new StableMapEntrySet(); }
+
+        @ForceInline
+        @Override
+        public V get(Object key) {
+            final StableValueImpl<V> stable = delegate.get(key);
+            if (stable == null) {
+                return null;
+            }
+            @SuppressWarnings("unchecked")
+            final K k = (K) key;
+            return stable.orElseSet(new Supplier<V>() {
+                @Override public V get() { return mapper.apply(k); }});
```

Comment:
This can return `null` (`StableMap` does allow `null` values), so the `getOrDefault` implementation in `AbstractImmutableMap` does not properly work for `StableMap`:
```java
var map = StableValue.map(Set.of(1), _ -> null);
// should print "null", but prints "default value"
System.out.println(map.getOrDefault(1, "default value"));
```

Reply:
Thanks for identifying this issue.